src/emailservice/email_server.rb

```diff
@@ -12,11 +12,40 @@
 
 post "/send_order_confirmation" do
   data = JSON.parse(request.body.read, object_class: OpenStruct)
-  Pony.mail(
-    to:       data.email,
-    from:     "noreply@example.com",
-    subject:  "Your confirmation email",
-    body:     erb(:confirmation, locals: { order: data.order }),
-    via:      :logger
-  )
+
+  # get the current auto-instrumented span
+  current_span = OpenTelemetry::Trace.current_span
+  current_span.add_attributes({
+    "app.order.id" => data.order.order_id,
+    "app.shipping.tracking.id" => data.order.shipping_tracking_id,
+    "app.shipping.cost.currency" => data.order.shipping_cost.currency_code,
+    "app.shipping.cost" => data.order.shipping_cost.units.to_s + "." + 
+      data.order.shipping_cost.nanos.to_s
+  })
+
+  send_email(data)
+
+  rescue Exception => e
+    # record exception in span (will create a span event)
+    current_span.record_exception(e)
+    raise e
+end
+
+def send_email(data)
+  # create and start a manual span
+  tracer = OpenTelemetry.tracer_provider.tracer('')
```

Comment:
Providing the empty string here is equivalent to not passing any arguments at all, and in both cases we log a warning when this happens:

```
WARN -- : calling TracerProvider#tracer without providing a tracer name.
```

(as per spec, if I recall). So we should give our tracer a name:

```suggestion
  tracer = OpenTelemetry.tracer_provider.tracer('emailservice')
```

Reply:
> Regarding the tracer with an empty name I want to bring @puckpuck to the thread, because i believe it would be better if we follow the same approach in all services.

I did some digging into this, and came to the conclusion we should use a name for each tracer since some SDKs will log an error or warning when an empty string is used.  I updated the other PRs to use the service name when getting the default tracer. I think we should accept this suggestion and use `emailservice` here.